ProjFS.Mac/PrjFSKextTests/TestMemory.cpp

```diff
@@ -0,0 +1,11 @@
+#include "../PrjFSKext/Memory.hpp"
+
```

Comment:
Much opportunity for fanciness here too, but that can be added later.

Reply:
I don't have a great solution for this yet, but I'd like to redefine `assert()` in test(able) code to call down to the XCTest equivalent. The XCTest asserts are a bit awkward because they expect `self` in the surrounding scope to refer to the test case object. I suspect we might want to store the current test in a thread-local variable or so and use that in some custom macros instead. Using regular `assert()` will currently just take down the whole process, which isn't quite as ergonomic with regard to pinpointing test failures.

Reply:
Ok thanks!